### ISSUE TITLE
tectonic installer bare metal - node error

### ISSUE THREAD

**Issue by stevemartinaws** [coreos/tectonic-forum#193]
### Tectonic Version 
1.6.4-tectonic.1
### Environment
running matchbox & dnsmasq in docker
tectonic install 1.6.4-tectonic.1 running in docker
deploying to HyperV VMs

### Expected Behavior
Run through installer
Boot Nodes
CoreOS deployed
Tectonic cluster created
### Actual Behavior
Run through installer
Boot Nodes
CoreOS seems to deploy however there is an error on each node and they seem to hang there. The installer keeps logging indefinitely.
![image](https://user-images.githubusercontent.com/16837023/30587520-5c92e51a-9d2b-11e7-8a1a-611ab4345006.png)

![image](https://user-images.githubusercontent.com/16837023/30587249-641535d2-9d2a-11e7-8730-3dd7b08959ad.png)
![image](https://user-images.githubusercontent.com/16837023/30589153-37f75604-9d31-11e7-8cfb-9bd8e36f7313.png)

![image](https://user-images.githubusercontent.com/16837023/30587147-0f2e7204-9d2a-11e7-933a-13cf6686a3ea.png)

### Reproduction Steps

  1. Run Tectonic Installer
  2. Power on nodes

### Other Information
Deploying 5 nodes (3 master, 2 worker)
DHCP and PXE boot environment seems to be working well, I can get DNS resolved properly

### Environment

What hardware/cloud provider/hypervisor is being used with Tectonic?

**Comment by zendeavor** [comment]
The screenshots of errors are certainly ugly, but I think not fatal. I saw similar ones in my efforts that came and went with no rhyme or reason, unless that's a kernel panic.

As for the installer hanging on the null_resource.* steps, further debugging would be performed by logging in to each node with SSH and checking `journalctl -xe`, `journalctl -u bootkube` (or `kubeconfig`?) and `docker logs <container id>`.

**Comment by stevemartinaws** [comment]
After rebooting a node with PXE boot i was able to connect and run `journalctl -xe` this has shown some errors

```Sep 19 13:23:31 SATECK8S02.sa.local kernel: hv_balloon: Received INFO_TYPE_MAX_PAGE_CNT
Sep 19 13:23:31 SATECK8S02.sa.local kernel: hv_balloon: Data Size is 8
Sep 19 13:23:32 SATECK8S02.sa.local update_engine[742]: I0919 13:23:32.600612   742 update_attempter.cc:493] Updating boot flags...
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1091]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1091]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1100]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1098]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1096]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1095]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1099]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1097]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1099]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1097]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1098]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1099]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1096]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1091]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1100]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.
Sep 19 13:23:32 SATECK8S02.sa.local systemd-udevd[1095]: Process '/usr/lib/udev/scripts/iscsidev.sh' failed with exit code 1.```

**Comment by zendeavor** [comment]
You're using an old installer version, and perhaps an old CoreOS as well? Try running the latest stables.

**Comment by stevemartinaws** [comment]
I have switched to using the latest coreos version and installer version. This seems to have gotten rid of the kernal panic and now my nodes seem to be stuck in a PXE boot - coreos - reboot loop.

I stopped my master nodes from pxe booting and they seem to have an issue booting into coreos...
![image](https://user-images.githubusercontent.com/16837023/30615090-99a5a820-9d85-11e7-9c0e-2956cea7f68b.png)
![image](https://user-images.githubusercontent.com/16837023/30615138-c340915e-9d85-11e7-8bed-1f85c8773490.png)

**Comment by zendeavor** [comment]
If your PXE environment is correctly configured, the initial boot from the network will install CoreOS to disk, then reboot to that on-disk install. At that point, the node will request an IP from your DHCP server, which should be accessible by the hostname you defined during the installer setup. It will then request its configurations from the matchbox host you also defined, install necessary cluster software, and start your tectonic console.

Your screenshots have truncated the errors, so I can only guess at the problem. My first intuition says your hosts are not accessible by their expected hostnames.

**Comment by rushins** [comment]
how is your disks configured (RAID 0 or RAID 5 )?  did you update latest firmware of disks before starting install? not sure about i see from your screen you have 2 issues ? disks and DHCP ? can you copy past your DHCP configuration and your matchbox setup with what version images in ASSETS folder of matchbox.

**Comment by stevemartinaws** [comment]
I have redeployed all the machines and attempted again - I am no longer seeing the issues above and CoreOS deploys over the network then boots into the local disk - which is what i expected. From this point nothing happens and the output from the masters and workers are shown below.

I get the feeling that the installer cannot connect or see the machines, I have able to ssh into them all with the key i supplied the installed via IP and DNS. The docker0 link issue may be related. Any thoughts?

**Masters:**
![image](https://user-images.githubusercontent.com/16837023/30697436-e38dbfb6-9ed6-11e7-95e6-9077befe9b2a.png)

**Workers**
![image](https://user-images.githubusercontent.com/16837023/30697494-1806aef6-9ed7-11e7-88e0-0c829d124407.png)

**Comment by zendeavor** [comment]
Troubleshoot now with the above mentioned steps, checking the kubectl logs, journalctl, and logs of docker containers.

**Comment by stevemartinaws** [comment]
journalctl reports that etcd failed to start on the master

``` Sep 21 13:28:23 SATECK8S01.sa.local sshd[1246]: Connection closed by 192.168.0.54 port 59368 [preauth]
Sep 21 13:28:23 SATECK8S01.sa.local etcd-wrapper[1241]: run: stat of host path /etc/ssl/etcd: stat /etc/ssl/etcd: no such file or directory
Sep 21 13:28:23 SATECK8S01.sa.local systemd[1]: etcd-member.service: Main process exited, code=exited, status=254/n/a
Sep 21 13:28:23 SATECK8S01.sa.local systemd[1]: Failed to start etcd (System Application Container).
-- Subject: Unit etcd-member.service has failed
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit etcd-member.service has failed.
--
-- The result is failed.
Sep 21 13:28:23 SATECK8S01.sa.local systemd[1]: etcd-member.service: Unit entered failed state.
Sep 21 13:28:23 SATECK8S01.sa.local systemd[1]: etcd-member.service: Failed with result 'exit-code'.
```

Nothing from the rest :
```
core@SATECK8S01 ~ $ journalctl -u bootkube
-- No entries --
core@SATECK8S01 ~ $ journalctl -u kubeconfig
-- No entries --
core@SATECK8S01 ~ $ docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
```

**Comment by zendeavor** [comment]
Did terraform log anything interesting? Did you enable ssh-agent on the local machine running the installer?

**Comment by stevemartinaws** [comment]
Turns out it was an authentication issue, which I have resolved by ensuring the installer is running under the correct user and the corresponding ssh keys are supplied to it. I do now have another issue with the tectonic console starting however I will log it separately. Thank you for all the help.

**Comment by tiagoaquino** [comment]
I am having same problem as you: 
On master node says "run: stat of host path /etc/ssl/etcd: stat /etc/ssl/etcd: no such file or directory". 
On worker node says: "invalid kubeconfig: stat /etc/kubernetes/kubeconfig: no such file or directory".

Any idea what can be? Thanks
